### PR TITLE
Feature: Bootstrap edge region

### PR DESCRIPTION
**Pull request by lukasbals**
## Summary of the changes / Why this is an improvement

Feature: Bootstrap edge region.

On hold since the backend endpoints need to be released first.

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: croud/regions/commands.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            print_response(
+                data=install_token,
+                errors=install_token_errors,
+                keys=["token"],
+                output_fmt=get_output_format(args),
+            )
+
+
+@require_confirmation(
+    "The generation of a deployment manfiest for an edge region is an experimental feature. Do you really want to use it?",  # noqa
```
Comment (matthijskrul) [inline croud/regions/commands.py:100]:
```suggestion
    "Generating a deployment manifest for an edge region is an experimental feature. Are you sure you want to proceed?",  # noqa
```

File: croud/regions/commands.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        output_fmt=get_output_format(args),
+    )
+
+    if not region_errors and region and "name" in region:
+        install_token, install_token_errors = client.get(
+            f"/api/v2/regions/{region['name']}/install-token/"
+        )
+
+        if not install_token_errors and install_token and "token" in install_token:
+            print_success("You successfully created a region.")
```
Comment (matthijskrul) [inline croud/regions/commands.py:83]:
```suggestion
            print_success("You have successfully created a region.")
```

File: croud/regions/commands.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    print_response(
+        data=data,
+        errors=errors,
+        keys=["name", "description"],
+        output_fmt=get_output_format(args),
+    )
+
+
+@require_confirmation(
+    "The creation of a region is an experimental feature. Do you really want to use it?",  # noqa
```
Comment (matthijskrul) [inline croud/regions/commands.py:45]:
```suggestion
    "Creating a region is an experimental feature. Are you sure you want to proceed?",  # noqa
```

File: tests/commands/test_regions.py [hunk 4]
```diff
@@ -27,3 +29,134 @@
...
+            "regions",
+            "create",
+            "--description",
+            "region-description",
+            "--provider",
+            "EDGE",
+        )
+    mock_request.assert_not_called()
+    mock_input.assert_called_once_with(
+        "The creation of a region is an experimental feature. Do you really want to use it? [yN] "  # noqa
```
Comment (matthijskrul) [inline tests/commands/test_regions.py:122]:
```suggestion
        "Creating a region is an experimental feature. Are you sure you want to proceed? [yN] "  # noqa
```

File: tests/commands/test_regions.py [hunk 5]
```diff
@@ -27,3 +29,134 @@
...
+        call_command(
+            "croud",
+            "regions",
+            "generate-deployment-manifest",
+            "--region-name",
+            "region-name",
+        )
+    mock_request.assert_not_called()
+    mock_input.assert_called_once_with(
+        "The generation of a deployment manfiest for an edge region is an experimental feature. Do you really want to use it? [yN] "  # noqa
```
Comment (matthijskrul) [inline tests/commands/test_regions.py:158]:
```suggestion
        "Generating a deployment manifest for an edge region is an experimental feature. Are you sure you want to proceed?" [yN] "  # noqa
```

**Review by matthijskrul — APPROVED** [review]

File: croud/regions/commands.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+        if not install_token_errors and install_token and "token" in install_token:
+            print_success("You have successfully created a region.")
+
+            print_raw(
+                [
+                    "",
+                    "To install the edge region run the following command:",
+                    "",
+                    f"  $ bash <( wget -qO- {client.base_url}/edge/{region['name']}/cratedb-cloud-edge.sh) {install_token['token']}",  # noqa
```
Comment (SStorm) [inline croud/regions/commands.py:89]: I think the link doesn't have the region name in it any more?

Comment (SStorm) [reply croud/regions/commands.py:89]: Actually I wonder if we can return the link from the API somehow :thinking:

Comment (lukasbals) [reply croud/regions/commands.py]: I think that would be nice. Let's create a story proposal for that?

**Review by SStorm — APPROVED** [review]
Nice!

**Pull request merged**